src/shared_gui_components/Component.cpp

```diff
@@ -464,157 +479,172 @@ void Component::createAbridgedLayout() {
 }
 
 void Component::createCompleteLayout() {
-  QLabel* label = nullptr;
-
-  QString string;
 
   auto* mainLayout = new QVBoxLayout();
 
-  QTableWidget* tableWidget = nullptr;
-
   ///! Error
   if (m_error) {
-    label = new QLabel("Errors");
+    auto* label = new QLabel("Errors");
     label->setObjectName("H1");
     mainLayout->addWidget(label);
 
     label = new QLabel(m_error->c_str());
     mainLayout->addWidget(label);
 
+    // TODO: why this empty label?
     label = new QLabel();
     mainLayout->addWidget(label);
   }
 
-  ///! Attributes
-  ///! Class BCL only stores double (optional units),
-  ///! int (optional units), and string, with their names.
-  //if (!m_attributes.empty()){
-  label = new QLabel("Attributes");
-  label->setObjectName("H1");
-  mainLayout->addWidget(label);
-
-  tableWidget = new QTableWidget(0, 2, this);
-  // really don't want the user to select or give focus to any table cells
-  tableWidget->setEditTriggers(QAbstractItemView::NoEditTriggers);
-  tableWidget->setSelectionMode(QAbstractItemView::NoSelection);
-  tableWidget->setAlternatingRowColors(true);
-  tableWidget->verticalHeader()->hide();
-  tableWidget->horizontalHeader()->hide();
-  tableWidget->setVerticalScrollBarPolicy(Qt::ScrollBarAlwaysOff);
-  tableWidget->setHorizontalScrollBarPolicy(Qt::ScrollBarAlwaysOff);
-  tableWidget->setSizePolicy(QSizePolicy::Preferred, QSizePolicy::Fixed);
-  tableWidget->horizontalHeader()->setSectionResizeMode(QHeaderView::ResizeToContents);
-  tableWidget->horizontalHeader()->setStretchLastSection(true);
-
-  mainLayout->addWidget(tableWidget);
+  // Description
+  if (!m_description.isEmpty()) {
+    {
+      auto* label = new QLabel("Description");
```

Comment:
Might want to show the section even if it's empty?

Reply:
I think it looks kinda weird when I do that

Reply:
Sounds good